crates/swc_plugin_runner/src/lib.rs

```diff
@@ -116,10 +163,10 @@ fn load_plugin(plugin_path: &Path, cache: &mut Option<PluginCache>) -> Result<In
                 // Generate an `ImportObject` from wasi_env
                 let mut import_object = wasi_env.import_object(&module)?;
 
-                // Inject `memory` into generated import_object, this is required to allocate
-                // guest (plugin)'s memory space from the host (SWC).
+                // This'll inject few interfaces into plugin's namespace to let plugin
+                // communicate with host
                 let mut env = Exports::new();
-                env.insert("memory", memory);
+                env.insert("__set_transform_result", set_transform_result_fn_decl);
```

Comment:
instead of expose memory and host deals with creating instance, now we pass environment to declared fn and let wasmer runtime sets memory accordingly. `set_transform_result` can access guest memory instance when guest calls it.